MzML/MzmlMethods.cs

```diff
@@ -140,34 +299,54 @@ public static void CreateAndWriteMyMzmlWithCalibratedSpectra(IMsDataFile<IMsData
             // Loop over all spectra
             for (int i = 1; i <= myMsDataFile.NumSpectra; i++)
             {
+
                 mzML.run.spectrumList.spectrum[i - 1] = new Generated.SpectrumType()
                 {
-                    defaultArrayLength = myMsDataFile.GetOneBasedScan(i).MassSpectrum.Size,
+                    defaultArrayLength = myMsDataFile.GetOneBasedScan(i).MassSpectrum.YArray.Length,
+                    index = (i - 1).ToString(CultureInfo.InvariantCulture),
+                    id = "scan=" + (myMsDataFile.GetOneBasedScan(i).OneBasedScanNumber).ToString(),
+                    cvParam = new Generated.CVParamType[10]
+                };
 
-                    index = i.ToString(CultureInfo.InvariantCulture),
-                    id = myMsDataFile.GetOneBasedScan(i).OneBasedScanNumber.ToString(),
+                //precursor info
+                mzML.run.spectrumList.spectrum[i - 1].precursorList = new Generated.PrecursorListType()
+                {
+                    count = 1.ToString()
 
-                    cvParam = new Generated.CVParamType[8]
                 };
-                mzML.run.spectrumList.spectrum[i - 1].cvParam[0] = new Generated.CVParamType();
+
+                // mzML.run.spectrumList.spectrum[i - 1].cvParam[0] = new Generated.CVParamType();
 
                 if (myMsDataFile.GetOneBasedScan(i).MsnOrder == 1)
                 {
-                    mzML.run.spectrumList.spectrum[i - 1].cvParam[0].accession = "MS:1000579";
+                    mzML.run.spectrumList.spectrum[i - 1].cvParam[0] = new Generated.CVParamType()
+                    {
+                        accession = "MS:1000579",
+                        cvRef = "MS",
+                        name = "MS1 spectrum",
+                        value = ""
+                    };
                 }
                 else if (myMsDataFile.GetOneBasedScan(i) is IMsDataScanWithPrecursor<IMzSpectrum<IMzPeak>>)
                 {
                     var scanWithPrecursor = myMsDataFile.GetOneBasedScan(i) as IMsDataScanWithPrecursor<IMzSpectrum<IMzPeak>>;
-                    mzML.run.spectrumList.spectrum[i - 1].cvParam[0].accession = "MS:1000580";
-
+                    mzML.run.spectrumList.spectrum[i - 1].cvParam[0] = new Generated.CVParamType
+                    {
+                        accession = "MS:1000580",
+                        cvRef = "MS",
+                        name = "MSn spectrum",
+                        value = ""
+                    };
+                    string precursorID = mzML.run.spectrumList.spectrum[i - 1].id;
```

Comment:
needs to be the precursor scan id, not id of the current scan. 
Used to be 
'
string precursorID = scanWithPrecursor.OneBasedPrecursorScanNumber.ToString();
'